proof/verifier.go

```diff
@@ -184,67 +188,273 @@ func (p *Proof) verifySplitRootProof() error {
 func (p *Proof) verifyExclusionProofs() (*commitment.TapCommitmentVersion,
 	error) {
 
-	// Gather all P2TR outputs in the on-chain transaction.
-	p2trOutputs := make(map[uint32]struct{})
+	// There is nothing to check so return early.
+	if p.ExclusionProofs == nil {
```

Comment:
I think this is incorrect. Because that would mean I can just create a proof with no exclusion proof and I wouldn't have to prove anything.
The code below proves that we actually have an exclusion proof **per p2tr output**. So the exit early would only be if `len(p2trOutputs) == 0`.